### ISSUE TITLE
Wrong units in inode usage graph

### ISSUE THREAD

**Issue by laulicus** [munin-monitoring/munin#916]
Munin master - commit id 84e3c1a5603c7c30229efed3bd8bed036471fffa

The graph shows m and u units instead of percents.

![munin-inode-usage](https://user-images.githubusercontent.com/28779954/36848408-7237039a-1d61-11e8-940f-a3e69cb6ad6c.png)

**Comment by sumpfralle** [comment]
Interesting!
Your graph looks like there was a change around the 24th of February. I committed a change to this plugin on that day (8a79dbffc1234b501d0c27e672a6212ace12c671). Could this be related?

What is the output of `munin-run df_inode config`? Does it contain the line `graph_scale no`? This should prevent the above value scaling.

Since there are multiple `df_inode` plugin implementations (for different platforms): which one do you use?

**Comment by laulicus** [comment]
I forget to mention, that I copied an old rrd file from 2.x to test it under 3.x. That's why the part 23-28 Febr data is missing.

http://demo.munin-monitoring.org/munin-monitoring.org/demo.munin-monitoring.org/df_inode.html uses 2.999.2-debianauto-2016-02-10-c493-gac4f73d and the units are correct. I revert to commit id cf02631c29f1b0512977f9d8a783522ab3e0b5db (2016-02-10) and try again, but the problem remains.

I tested with Docker using Alpine and Debian Stretch too. Does it help if I attach the Dockerfile?

**Comment by sumpfralle** [comment]
Thank you for the details!

I still have no idea, what could be wrong.

> I tested with Docker using Alpine and Debian Stretch too. Does it help if I attach the Dockerfile?

I am not famiiiar with docker - but maybe someone else wants to chime in?

Could you please take a look at my questions below?

> What is the output of munin-run df_inode config? Does it contain the line "graph_scale no"? This should prevent the above value scaling.
>
> Since there are multiple "df_inode" plugin implementations (for different platforms): which one do you use?

**Comment by laulicus** [comment]
munin-run df_inode config
graph_scale no

df_inode plugin comes from Debian 9 packages munin-plugins-core 2.0.33-1.

With munin 2.0.6-4+deb7u2~bpo60+1 it works, but not with the current master.

**Comment by niclan** [comment]
graph_scale no should be enough to avoid the milli-percentages. I commented this in the patch too.

**Comment by laulicus** [comment]
So the graph_scale settings is correct. Have you any other idea, why it shows bad unit?

**Comment by niclan** [comment]
Can you please show the complete output of "munin-run df_ifnode confg" from the problematic node? (not on the server, on the node whose graph is showing milli percent).

What does the load plugin graph look like?

```
# munin-run load config
graph_title Load average
graph_args --base 1000 -l 0
graph_vlabel load
graph_scale no
graph_category system
load.label load
graph_info The load average of the machine describes how many processes are in the run-queue (scheduled to run "immediately").
load.info 5 minute load average
```

**Comment by laulicus** [comment]
```
# munin-run df_inode config
graph_title Inode usage in percent
graph_args --upper-limit 100 -l 0
graph_vlabel %
graph_scale no
graph_category disk
_dev.label /dev
_dev.warning 92
_dev.critical 98
_run.label /run
_run.warning 92
_run.critical 98
_dev_xvda1.label /
_dev_xvda1.warning 92
_dev_xvda1.critical 98
_dev_shm.label /dev/shm
_dev_shm.warning 92
_dev_shm.critical 98
_run_lock.label /run/lock
_run_lock.warning 92
_run_lock.critical 98
_sys_fs_cgroup.label /sys/fs/cgroup
_sys_fs_cgroup.warning 92
_sys_fs_cgroup.critical 98

# munin-run df_inode
_dev.value 0.130106378771701
_run.value 0.278616933022368
_dev_xvda1.value 6.56044006347656
_dev_shm.value 0.000391315917166247
_run_lock.value 0.00117394775149874
_sys_fs_cgroup.value 0.0058697387574937

# munin-run load config
graph_title Load average
graph_args --base 1000 -l 0
graph_vlabel load
graph_scale no
graph_category system
load.label load
graph_info The load average of the machine describes how many processes are in the run-queue (scheduled to run "immediately").
load.info 5 minute load average
```

**Comment by sumpfralle** [comment]
I guess, now we can be sure, that it is a issue with munin 2.999?

**Comment by laulicus** [comment]
Yes.

**Comment by niclan** [comment]
I agree. But to be more sure @laulicus : does your load graph also have milli load?

**Comment by laulicus** [comment]
Yes, `munin-run load` shows 0.19, but the graph shows 190.00m.

**Comment by niclan** [comment]
Pretty sure this is 2.999 then. I would class it as major since it affects the readability of graphs.
